UnityProject/Assets/Scripts/Robotics/Bots/InteractableBot.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System.Collections;
+using UnityEngine;
+using Mirror;
+using Systems.MobAIs;
+
+namespace Robotics
+{
+	public class InteractableBot : NetworkBehaviour, IPredictedCheckedInteractable<HandApply>
+	{
+		HandApply interaction;
+
+		[SerializeField] private SpriteRenderer SPRITE_RENDERER;
+		[SerializeField] private Sprite EMAGGED_SPRITE;
+
+		private MobExplore mobController;
+		public MobExplore MobController
+		{
+			get
+			{
+				if (!mobController)
+				{
+					mobController = GetComponent<MobExplore>();
+				}
+				return mobController;
+			}
+		}
+		public void ClientPredictInteraction(HandApply interaction) { }
+
+		public void ServerRollbackClient(HandApply interaction) { }
+
+		public bool WillInteract(HandApply interaction, NetworkSide side)
+		{
+			if (!DefaultWillInteract.Default(interaction, side)) return false;
+			if (interaction.TargetObject != gameObject) return false;
+			if (interaction.HandObject != null && interaction.Intent == Intent.Harm) return false;
+
+			return MobController != null;
+		}
+		public void ServerPerformInteraction(HandApply interaction)
+		{
+			this.interaction = interaction;
+
+			if (Validations.HasItemTrait(interaction.HandObject, CommonTraits.Instance.Emag)
+				&& interaction.HandObject.TryGetComponent<Emag>(out var emag)
+				&& emag.EmagHasCharges())
+			{
+				TryEmag(emag, interaction);
+			}
+		}
+		public void TryEmag(Emag emag, HandApply interaction)
```

Comment:
the name of this method suggest it will attempt to do something that could fail or that it can ``out`` a reference and returns a bool.

Reply:
This can fail if the Emag has no more charges, can still rename if requested.